### PR TITLE
CF_Http: PLATFORM-1059 - retry all HTTP 50x responses

### PR DESCRIPTION
**Pull request by macbre**
https://wikia-inc.atlassian.net/browse/PLATFORM-1059

Retry all HTTP 50x responses from DFS

Extend 1a0579bd338c749ae45859596df7c933e5a43655

@michalroszka

### CODE REVIEW

File: extensions/SwiftCloudFiles/php-cloudfiles-1.7.10/cloudfiles_http.php [hunk 1]
```diff
@@ -1483,7 +1483,8 @@ private function _send_request($conn_type, $url_path, $hdrs=NULL, $method="GET",
 		while( $retriesLeft >= 0 ) {
 			$res = $this->_do_send_request( $conn_type, $url_path, $hdrs, $method, $force_new );
 
-			if ( !in_array( $res, [ 500, 503, false ] ) ) {
+			# PLATFORM-1059 - retry all HTTP 50x responses
+			if ( $res !== false && $res < 500 ) {
 				// request was successful, return
```
Comment (ghost) [inline extensions/SwiftCloudFiles/php-cloudfiles-1.7.10/cloudfiles_http.php]: The comment here is slighly misleading. Note, that 4xx indicate client errors. We can't say the request successful on client error. :-)

Comment (macbre) [inline extensions/SwiftCloudFiles/php-cloudfiles-1.7.10/cloudfiles_http.php]: HTTP 40x responses are fine - for instance 404 on HEAD request means that a given file does not exist.

We only want to perform a retry on the backend error.

Comment (macbre) [inline extensions/SwiftCloudFiles/php-cloudfiles-1.7.10/cloudfiles_http.php]: @michalroszka - the comment improved in 64f8354

File: extensions/SwiftCloudFiles/php-cloudfiles-1.7.10/cloudfiles_http.php [hunk 2]
```diff
@@ -1483,8 +1483,9 @@ private function _send_request($conn_type, $url_path, $hdrs=NULL, $method="GET",
 		while( $retriesLeft >= 0 ) {
 			$res = $this->_do_send_request( $conn_type, $url_path, $hdrs, $method, $force_new );
 
-			if ( !in_array( $res, [ 500, 503, false ] ) ) {
-				// request was successful, return
+			# PLATFORM-1059 - retry all HTTP 50x responses
+			if ( $res !== false && $res < 500 ) {
+				// request was successful (i.e. DFS backend did not fail), return
```
Comment (ghost) [inline extensions/SwiftCloudFiles/php-cloudfiles-1.7.10/cloudfiles_http.php]: Yeah, not it slightly less misleading. :-) I would actually remove it and leave "Retry on server errors (HTTP 50x responses)." :8ball:

Comment (macbre) [inline extensions/SwiftCloudFiles/php-cloudfiles-1.7.10/cloudfiles_http.php]: Removed in 87dc70c

**Comment by ghost** [comment]
:+1:

**Pull request merged**